hnn_core/network_builder.py

```diff
@@ -557,12 +582,19 @@ def get_data_from_neuron(self):
             isoma_py[gid] = {key: rec_i.to_python()
                              for key, rec_i in rec_i.items()}
 
+        rec_arr_py = dict()
+        rec_times_py = dict()
+        for arr_name, nrn_arr in self._rec_array.items():
+            rec_arr_py.update({arr_name: nrn_arr._get_nrn_voltages()})
+            rec_times_py.update({arr_name: nrn_arr._get_nrn_times()})
+
         from copy import deepcopy
         data = (self._all_spike_times.to_python(),
                 self._all_spike_gids.to_python(),
                 deepcopy(self.net.gid_ranges),
                 deepcopy(vsoma_py),
-                deepcopy(isoma_py))
+                deepcopy(isoma_py),
+                rec_arr_py, rec_times_py)
```

Comment:
Any reason not to deepcopy? Although truthfully I've forgotten the rationale for deepcopying isoma/vsoma.

Reply:
I'm 99% certain there's no reason for deepcopy here. The output dicts are created inside the function, so there's no chance they'll get mixed up. Right @jasmainak ?

Reply:
You should use `deepcopy` when you want to be safe. However, the flipside is that there will be a performance hit. Often during profiling, the `copy` methods are to blame (for memory and time)

Reply:
I think we should think about performance! `get_data_from_neuron` is called exactly once for every call to `_clone_and_simulate`, which immediately returns. The built `neuron_net` dies inside `_clone_and_simulate` and gets garbage collected.

Having 'safety calls` `deepcopy` seems like a bad habit. Surely we should rather write better test? ;)

Reply:
Agree we should write more and better tests :)

Performance will be a consideration here if we do multiple trials or many sequential calls to `simulate_dipole` for example during parameter optimization or grid search